lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -27,6 +31,205 @@ class WP_Theme_JSON_6_1 extends WP_Theme_JSON_6_0 {
 		'h6'     => 'h6',
 		'button' => '.wp-element-button, .wp-block-button__link', // We have the .wp-block-button__link class so that this will target older buttons that have been serialized.
 	);
+
+		/**
+		 * Sanitizes the input according to the schemas.
+		 *
+		 * @since 5.8.0
+		 * @since 5.9.0 Added the `$valid_block_names` and `$valid_element_name` parameters.
+		 *
+		 * @param array $input               Structure to sanitize.
+		 * @param array $valid_block_names   List of valid block names.
+		 * @param array $valid_element_names List of valid element names.
+		 * @return array The sanitized output.
+		 */
+	protected static function sanitize( $input, $valid_block_names, $valid_element_names ) {
+
+		$output = array();
+
+		if ( ! is_array( $input ) ) {
+			return $output;
+		}
+
+		// Preserve only the top most level keys.
+		$output = array_intersect_key( $input, array_flip( static::VALID_TOP_LEVEL_KEYS ) );
+
+		// Remove any rules that are annotated as "top" in VALID_STYLES constant.
+		// Some styles are only meant to be available at the top-level (e.g.: blockGap),
+		// hence, the schema for blocks & elements should not have them.
+		$styles_non_top_level = static::VALID_STYLES;
+		foreach ( array_keys( $styles_non_top_level ) as $section ) {
+			foreach ( array_keys( $styles_non_top_level[ $section ] ) as $prop ) {
+				if ( 'top' === $styles_non_top_level[ $section ][ $prop ] ) {
+					unset( $styles_non_top_level[ $section ][ $prop ] );
+				}
+			}
+		}
+
+		// Build the schema based on valid block & element names.
+		$schema                         = array();
+		$schema_styles_elements         = array();
+		$schema_styles_pseudo_selectors = array();
+
+		// Set allowed element pseudo selectors based on per element allow list.
+		foreach ( $valid_element_names as $element ) {
+			$schema_styles_elements[ $element ] = $styles_non_top_level;
+
+			if ( array_key_exists( $element, static::VALID_ELEMENT_PSEUDO_SELECTORS ) ) {
+				foreach ( static::VALID_ELEMENT_PSEUDO_SELECTORS[ $element ] as $selector ) {
+					$schema_styles_elements[ $element ][ $selector ] = $styles_non_top_level;
+				}
+			}
+		}
```

Comment:
Here we add a new sub item for each pseudo selector to the schema for each `element` (e.g. `link`, `h1`...etc). Each selector is the key with the valid style values as the value.